WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/IssueRefundViewModel.swift

```diff
@@ -444,14 +471,14 @@ extension IssueRefundViewModel {
     /// Calculates whether the next button should be animating or not
     ///
     private func calculateNextButtonAnimatingState() -> Bool {
-        // When we have a chargeID, we need to wait until we fetch the charge.
-        state.charge == nil && !state.order.chargeID.isNilOrEmpty
+        // When we have a chargeID, we need to wait until we fetch the charge unless there is an error fetching the charge.
+        state.charge == nil && !state.order.chargeID.isNilOrEmpty && state.fetchChargeError == nil
     }
 
     /// Calculates whether there are pending changes to commit
     ///
     private func calculatePendingChangesState() -> Bool {
-        state.refundQuantityStore.count() > 0 || state.shouldRefundShipping || state.shouldRefundFees
```

Comment:
Not a blocker, but what do you think of using a guard here? I think it would be a little bit more readable because that logical expression already has four subexpressions :) 

```
private func calculatePendingChangesState() -> Bool {
        guard state.fetchChargeError == nil else {
            return false
        }


        return state.refundQuantityStore.count() > 0 ||
               state.shouldRefundShipping ||
               state.shouldRefundFees
    }
```

Reply:
Good suggestion, it does read better this way! Updated in https://github.com/woocommerce/woocommerce-ios/pull/6719/commits/106b4e5bf39e733eff651200a371e72a19b5a03d